builtin/tfc/config_sourcer.go

```diff
@@ -0,0 +1,478 @@
+			Value: value,
+		}
+	}
+
+	return results, nil
+}
+
+func (cs *ConfigSourcer) stop() error {
+	cs.cacheMu.Lock()
+	defer cs.cacheMu.Unlock()
+
+	if cs.client != nil {
+		cs.client.CloseIdleConnections()
+		cs.client = nil
+	}
+
+	// Reset our results tracking to empty. This will force the next call
+	// to rebuild all our secret values.
+	var zeroTime time.Time
+	cs.lastRead = zeroTime
+	cs.secretCache = nil
+
+	return nil
+}
+
+func (cs *ConfigSourcer) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(
+		docs.FromConfig(&sourceConfig{}),
+		docs.RequestFromStruct(&reqConfig{}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description("Read configuration values from Terraform Cloud.")
+
+	doc.Example(`
+config {
+  env = {
+    "DATABASE_USERNAME" = configdynamic("terraform-cloud", {
+			organization = "foocorp"
+			workspace = "databases"
+			output = "db_username"
+    })
+
+    "DATABASE_PASSWORD" = configdynamic("vault", {
+			organization = "foocorp"
+			workspace = "databases"
+			output = "db_password"
+    })
+
+    "DATABASE_HOST" = configdynamic("vault", {
+			organization = "foocorp"
+			workspace = "databases"
+			output = "db_hostname"
+    })
+  }
+}
+`)
+
+	doc.SetRequestField(
+		"organization",
+		"The Terraform Cloud organization to query",
+		docs.Summary(
+			"within a single workspace, multiple dynamic values that use the same",
+			"organization and workspace will only read the value once. This allows outputs",
+			"to be extracted into multiple values. The example above shows",
+			"this functionality by reading the username and password into separate values.",
+		),
+	)
+
+	doc.SetRequestField(
+		"workspace",
+		"The Terraform Cloud workspace associated with the given organization to read the outputs of",
+		docs.Summary(
+			"The outputs associtaed with the most recent state version for the given workspace",
+			"are the ones that are used. These values are refreshed according to",
+			"refreshInternal, a source field.",
+		),
+	)
+
+	doc.SetRequestField(
+		"output",
+		"The name of the output to read the value of",
+	)
+
+	doc.SetField(
+		"token",
+		"The Terraform Cloud API token",
+		docs.Summary(
+			"The token is used to authenticate access to the specific organization and",
+			"workspace. Tokens are managed at https://app.terraform.io/app/settings/tokens.",
+		),
+	)
+
+	doc.SetField(
+		"base_url",
+		"The scheme, host, and port to calculate the URL to fetch using",
+		docs.Summary(
+			"This is provided to allow users to query values from Terraform Enterprise",
+			"installations",
+		),
+		docs.Default("https://api.terraform.io"),
+	)
+
+	doc.SetField(
+		"skip_verify",
+		"Do not validate the TLS cert presented by the Vault server.",
+		docs.Summary(
+			"This is not recommended unless absolutely necessary.",
+		),
+		docs.EnvVar("VAULT_SKIP_VERIFY"),
+	)
+
+	doc.SetField(
+		"refresh_interval",
+		"How often the outputs should be fetch.",
+		docs.Default("10m"),
```

Comment:
We could make this the actual variable just to avoid any mistakes if the interval changes and the docs aren't updated?

```suggestion
		docs.Default(refreshPeriod.String()),
```